### ISSUE TITLE
use a multidimensional array?

### ISSUE THREAD

**Issue by mvattuone** [mvattuone/conway-game-of-life-cpp#1]
some of this is kinda messy, especially keeping track of each and every cell.  

at the same time, I just... don't like for loops inside for loops. 🤷‍♂️

**Comment by poscar** [comment]
I think even with multidimensional array you'll have to do a 2d iteration. Unless you do matrix operations or bit shifting to accomplish the game behavior.

**Comment by mvattuone** [comment]
Yeah I am working on a similar thing in Javascript right now, and went from a two-dimensional to a one-dimensional when I decided to use a `Uint8Array` instead of a regular one, which... sucked, but did help w performance.
